lapce-proxy/src/dispatch.rs

```diff
@@ -1227,6 +1238,34 @@ fn git_get_remote_file_url(workspace_path: &Path, file: &Path) -> Result<String>
     Ok(url)
 }
 
+fn git_fetch(workspace_path: &Path) -> Result<()> {
+    let repo = Repository::discover(workspace_path)?;
+
+    for remote in repo.remotes()?.iter().flatten() {
+        let mut remote = repo.find_remote(remote)?;
+        let refspecs: Vec<String> = remote
+            .refspecs()
+            .filter_map(|r| Some(r.str()?.to_string()))
+            .collect();
+        let mut callbacks = RemoteCallbacks::new();
+        callbacks.credentials(|_url, username_from_url, _allowed_types| {
+            Cred::ssh_key(
+                username_from_url.unwrap(),
+                None,
+                std::path::Path::new(&format!(
+                    "{}/.ssh/id_rsa",
+                    env::var("HOME").unwrap()
+                )),
+                None,
```

Comment:
this isn't right:
1. you should not use `HOME` envvar
2. it doesn't have to be RSA key
3. key doesn't have to exist with name `id_{key_type}`
4. key doesn't have to exist as a file, it could be smartcard

Reply:
I am very aware that this isn't a perfect solution.  I was hoping to discuss some of these things.
I'm assume we are going to have to make the key file configurable?  The user might not even be using ssh.  I'll be looking into this more, but wanted to see if you or others had any ideas.

Reply:
> I'm assume we are going to have to make the key file configurable?

There should be nothing configurable, it's up to git to handle everything

Reply:
credential helper will handle whole authentication, there is no reason to handle auth any other way these days

Reply:
I was under the impression that credential helper only dealt with password-based auth, not ssh.
https://docs.rs/git2/latest/git2/struct.Cred.html#method.credential_helper

Reply:
It depends on the credential helper, but you can add ssh agent on credential helper failure